src/queries/labelText.ts

```diff
@@ -11,14 +14,38 @@ import type {
   QueryByQuery,
 } from './makeQueries';
 
+const matchAccessibilityLabelledByText = (
+  rootInstance: ReactTestInstance,
+  nativeID: string | undefined,
+  text: TextMatch
+) => {
+  if (!nativeID) {
+    return false;
+  }
+
+  return (
+    rootInstance.findAll(
+      (node) =>
+        filterNodeByType(node, Text) &&
+        matchStringProp(node.props.nativeID, nativeID) &&
+        getNodeByText(node, text)
+    ).length > 0
+  );
+};
+
 const queryAllByLabelText = (
   instance: ReactTestInstance
 ): ((text: TextMatch) => Array<ReactTestInstance>) =>
   function queryAllByLabelTextFn(text) {
     return instance.findAll(
       (node) =>
         typeof node.type === 'string' &&
-        matchStringProp(node.props.accessibilityLabel, text)
+        (matchStringProp(node.props.accessibilityLabel, text) ||
```

Comment:
Pls extract `matchAccessibilityLabel` function to encapsulate the `||` logic so that the condition here is simplier and contains only `&&` operatos?